src/components/wallets/WalletAccordion.tsx

```diff
@@ -0,0 +1,357 @@
+import { useState } from 'react'
+import { Accordion, AccordionSummary, AccordionDetails } from '@mui/material'
+import styled, { css } from 'styled-components'
+import { DateTime } from 'luxon'
+import { gql } from '@apollo/client'
+
+import { theme } from '~/styles'
+import {
+  Avatar,
+  Button,
+  Icon,
+  Skeleton,
+  Status,
+  StatusEnum,
+  Tooltip,
+  Typography,
+} from '~/components/designSystem'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { WalletDetailsFragment, WalletStatusEnum } from '~/generated/graphql'
+
+gql`
+  fragment WalletDetails on Wallet {
+    id
+    balance
+    consumedCredits
+    createdAt
+    creditsBalance
+    currency
+    expirationDate
+    lastBalanceSyncAt
+    lastConsumedCreditAt
+    name
+    rateAmount
+    status
+    terminatedAt
+  }
+`
+
+interface WalletAccordionProps {
+  wallet: WalletDetailsFragment
+}
+
+const mapStatus = (type?: WalletStatusEnum | undefined) => {
+  switch (type) {
+    case WalletStatusEnum.Active:
+      return {
+        type: StatusEnum.running,
+        label: 'text_62da6ec24a8e24e44f812874',
+      }
+    default:
+      return {
+        type: StatusEnum.failed,
+        label: 'text_62da6ec24a8e24e44f8128b0',
+      }
+  }
+}
+
+export const WalletAccordion = ({ wallet }: WalletAccordionProps) => {
+  const status = mapStatus(wallet?.status)
+  let [creditAmountUnit = '0', creditAmountCents = '00'] = wallet.creditsBalance.split('.')
+  let [consumerCreditUnit = '0', consumerCreditCents = '00'] = wallet.consumedCredits.split('.')
+  const { translate } = useInternationalization()
+  // All active wallets should be opened by default on first render
+  const [isOpen, setIsOpen] = useState(wallet?.status === WalletStatusEnum.Active)
+
```

Comment:
You're accessing a lot `wallet.` I guess you could spread it. No need to really change it, just a remark :)

Reply:
Yep ! Did it :)